src/main/java/com/newrelic/telemetry/opentelemetry/export/CommonUtils.java

```diff
@@ -0,0 +1,55 @@
+package com.newrelic.telemetry.opentelemetry.export;
+
+import com.newrelic.telemetry.Attributes;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.Map;
+
+// todo: this is a terrible name.
+public class CommonUtils {
+
+  static Attributes populateLibraryInfo(
+      Attributes attributes, InstrumentationLibraryInfo instrumentationLibraryInfo) {
+    if (instrumentationLibraryInfo != null) {
+      if (instrumentationLibraryInfo.getName() != null
+          && !instrumentationLibraryInfo.getName().isEmpty()) {
+        attributes.put("instrumentation.name", instrumentationLibraryInfo.getName());
+      }
+      if (instrumentationLibraryInfo.getVersion() != null
+          && !instrumentationLibraryInfo.getVersion().isEmpty()) {
+        attributes.put("instrumentation.version", instrumentationLibraryInfo.getVersion());
+      }
+    }
+    return attributes;
+  }
+
+  static Attributes addResourceAttributes(Attributes attributes, Resource resource) {
+    if (resource != null) {
+      Map<String, AttributeValue> labelsMap = resource.getAttributes();
+      putInAttributes(attributes, labelsMap);
+    }
+    return attributes;
+  }
+
+  static void putInAttributes(
+      Attributes attributes, Map<String, AttributeValue> originalAttributes) {
+    originalAttributes.forEach(
+        (key, value) -> {
+          switch (value.getType()) {
+            case STRING:
+              attributes.put(key, value.getStringValue());
+              break;
+            case LONG:
+              attributes.put(key, value.getLongValue());
+              break;
+            case BOOLEAN:
+              attributes.put(key, value.getBooleanValue());
+              break;
+            case DOUBLE:
+              attributes.put(key, value.getDoubleValue());
+              break;
+          }
+        });
+  }
+}
```

Comment:
There's a lot of logic in these helper methods, yet no unit test coverage.  I'd like to see coverage added.

Reply:
mind if we save that for a follow-on PR?  I can create an issue to track it.

Reply:
(it is actually covered by other tests already, just not ones just for this class)

Reply:
Sure!  That works.

Reply:
#29 filed.